datanode/api/trading_data_v2.go

```diff
@@ -3047,6 +3055,139 @@ func (t *tradingDataServiceV2) GetVegaTime(ctx context.Context, _ *v2.GetVegaTim
 }
 
 // -- NetworkHistory --.
+func (t *tradingDataServiceV2) ExportNetworkHistory(req *v2.ExportNetworkHistoryRequest, stream v2.TradingDataService_ExportNetworkHistoryServer) error {
+	defer metrics.StartAPIRequestAndTimeGRPC("ExportNetworkHistory")()
+
+	if req.Table == v2.Table_TABLE_UNSPECIFIED {
+		return formatE(ErrNetworkHistoryNoTableName, errors.New("empty table name"))
+	}
+
+	tableName := strings.TrimPrefix(strings.ToLower(req.Table.String()), "table_")
+
+	allSegments, err := t.networkHistoryService.ListAllHistorySegments()
```

Comment:
you are going to hate me......but....could you use the setup in the network history tests which will give you a fully setup service with segment data and use this in here so you could write an end to end test, confirming the file structure and contents?  or maybe this is overkill, the code does look fairly involved though so may be benefit

Reply:
OK I started doing that and it turned into quite a faff so I've made a unit test instead which does some basic sanity checking of the `ExportNetworkHistory` API function, mocking out the network history store.

I've run it end to end myself and have checked that it all works.